### PR TITLE
Updated react and react-dom. Added prop-types library.

### PR DESCRIPTION
**Pull request by DovydasNavickas**
- [x] Updated react and react-dom.
- [x] Added prop-types library.
- [x] Updated webpack to take externals from package.json.

Solves:
React PropTypes deprecation #2

### CODE REVIEW

File: src/abstractions/base-handler.ts [hunk 1]
```diff
@@ -44,9 +45,9 @@ export abstract class BaseHandler<TProps extends BaseHandlerProps, TState extend
     };
 
     static childContextTypes = {
-        DropdownOpen: React.PropTypes.bool,
-        DropdownOnHeaderClickCallback: React.PropTypes.func,
-        DropdownOnSectionClickCallback: React.PropTypes.func
+        DropdownOpen: PropTypes.bool,
```
Comment (DovydasNavickas) [inline src/abstractions/base-handler.ts]: Not sure whether `PropTypes.bool.isRequired` should be here, but if these properties are required, we'll have to update the prop-types provided here.
@MartynasZilinskas?

Comment (MartynasZilinskas) [inline src/abstractions/base-handler.ts]: I think it should, otherwise components will not work properly.

**Comment by MartynasZilinskas** [comment]
Please reword your commit by `commitizen` conventions.

**Comment by DovydasNavickas** [comment]
I've updated all prop-types to be required by adding `.isRequired` suffix.

:shipit: ?

**Comment by DovydasNavickas** [comment]
I'll rebase if everything is OK with the code.

**Comment by DovydasNavickas** [comment]
Rebased. Everything's OK, @MartynasZilinskas?

**Comment by MartynasZilinskas** [comment]
All good.

**Pull request merged**